src/game_events/manager.hpp

```diff
@@ -55,21 +57,23 @@ class manager
 	manager& operator=(const manager&) = delete;
 
 	explicit manager();
-	void read_scenario(const config& scenario_cfg);
+	void read_scenario(const config& scenario_cfg, game_lua_kernel& lk);
 	~manager();
 
-	/** Create an event handler. */
-	void add_event_handler(const config& handler, bool is_menu_item = false);
+	/** Create an event handler from an [event] tag. */
+	void add_event_handler_from_wml(const config& handler, game_lua_kernel& lk, bool is_menu_item = false);
+	/** Create an empty event handler. Expects the caller to finish setting up the event. */
+	pending_event_handler add_event_handler_from_lua(const std::string& name, const std::string& id, bool repeat = false, bool is_menu_item = false);
 
 	/** Removes an event handler. */
 	void remove_event_handler(const std::string& id);
 
 	/** Gets an event handler by ID */
 	const handler_ptr get_event_handler_by_id(const std::string& id);
 
-	void add_events(const config::const_child_itors& cfgs, const std::string& type = std::string());
+	void add_events(const config::const_child_itors& cfgs, game_lua_kernel& lk, const std::string& type = std::string());
 
-	void write_events(config& cfg) const;
+	void write_events(config& cfg, bool include_nonserializable=false) const;
```

Comment:
Needs a comment to explain what serializing a non-serializable thing will do.

Reply:
Thanks for adding the comment, but please make it Doxygen-style (as in, a Doxygen doc for the function).

Reply:
Still unresolved. On the same subject, please document the situation for `event_handler::write_config` too.